pkg/services/sqlstore/secretstore.go

```diff
@@ -0,0 +1,141 @@
+package sqlstore
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/events"
+	"github.com/grafana/grafana/pkg/infra/metrics"
+	"github.com/grafana/grafana/pkg/models"
+	"xorm.io/xorm"
+)
+
+// GetSecret adds a secret to the query model by querying by org_id as well as
+// either uid (preferred), id, or name and is added to the bus.
+func (ss *SQLStore) GetSecret(ctx context.Context, query *models.GetSecretQuery) error {
+	metrics.MDBSecretQueryByID.Inc()
+
+	return ss.WithDbSession(ctx, func(sess *DBSession) error {
+		if query.OrgId == 0 || (query.Id == 0 && len(query.EntityUid) == 0) {
+			return models.ErrSecretIdentifierNotSet
+		}
+
+		secret := &models.Secret{EntityUid: query.EntityUid, OrgId: query.OrgId, Id: query.Id}
+		has, err := sess.Get(secret)
+
+		if err != nil {
+			sqlog.Error("Failed getting data source", "err", err, "uid", query.EntityUid, "id", query.Id, "orgId", query.OrgId)
+			return err
+		} else if !has {
+			return models.ErrSecretNotFound
+		}
+
+		query.Result = secret
+
+		return nil
+	})
+}
+
+func (ss *SQLStore) GetSecrets(ctx context.Context, query *models.GetSecretsQuery) error {
+	var sess *xorm.Session
+	return ss.WithDbSession(ctx, func(dbSess *DBSession) error {
+		if query.SecretLimit <= 0 {
+			sess = dbSess.Where("org_id=?", query.OrgId).Asc("id")
+		} else {
+			sess = dbSess.Limit(query.SecretLimit, 0).Where("org_id=?", query.OrgId).Asc("id")
+		}
+
+		query.Result = make([]*models.Secret, 0)
+		return sess.Find(&query.Result)
+	})
+}
+
+// DeleteSecret removes a secret by org_id as well as either uid (preferred), id, or name
+// and is added to the bus.
+func (ss *SQLStore) DeleteSecret(ctx context.Context, cmd *models.DeleteSecretCommand) error {
```

Comment:
Does it make sense to break this out into DeleteSecretByOrgId and DeleteSecretByUid ?